packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +37,140 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+//get data from custom polcies file and write custom policies to CFN template
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceName: string,
+  service: string,
+  resourceDir: string,
+  cfnFile: string,
+  category: string,
+  filePath: string
+) {
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = stateManager.getCustomPolicies(service, category, resourceName);
+  if (!customPolicies || !this.validateCustomPoliciesSchema(customPolicies)) {
+    return;
+  }
+
+  await addCustomPoliciesToCFNTemplate(service, customPolicies.policies, cfnTemplate, filePath, resourceName, {templateFormat} );
+
+}
+
+//merge the custom IAM polciies to CFN template for lambda and API container
+
+export async function addCustomPoliciesToCFNTemplate(
```

Comment:
shouldn't be exported. Also applies to the functions below that are only used in this file